hathor/transaction/base_transaction.py

```diff
@@ -763,6 +795,23 @@ def start_mining(self, start: int = 0, end: int = MAX_NONCE, sleep_seconds: floa
                     return None
         return None
 
+    def set_height_hint(self, height: int) -> None:
+        """This method exists to set the height metadata when we can't calculate it yet (i.e. syncing checkpoints).
+
+        `TxValidationError` will be risen if a height already exists (which is not determined by its value but by the
+        validation state, that is, a height of `None` on a full validated transaction is final) and it doesn't match
+        the given hint. This is in place in order to prevent accidentally setting the height to a wrong value after the
+        value has been validated, the height must be checked when basic-validating this tx and effort should be made to
+        always set the correct height when using this method.
+        """
+        # TODO: all the checks and validations mentioned on the docstring above
+        assert self.storage is not None
+        assert self.hash is not None
+        metadata = self.storage.get_metadata(self.hash)
+        if metadata is None:
```

Comment:
Is it possible to happen?

Reply:
Actually I don't think that it can be not `None` because by the point that we run `set_height_hint` the block wasn't even saved for the first time.

Reply:
Old conversation. It does make sense anymore. Marking as resolved.